manager/integration/tests/test_engine_upgrade.py

```diff
@@ -747,61 +750,272 @@ def test_engine_live_upgrade_with_intensive_data_writing(client, core_api, volum
     assert volume_file_md5sum2 == original_md5sum2
 
 
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_auto_upgrade_engine_to_default_version():
+def prepare_auto_upgrade_engine_to_default_version(client): # NOQA
+    default_img = common.get_default_engine_image(client)
+    default_img_name = default_img.name
+    default_img = wait_for_engine_image_ref_count(client, default_img_name, 0)
+    cli_v = default_img.cliAPIVersion
+    cli_minv = default_img.cliAPIMinVersion
+    ctl_v = default_img.controllerAPIVersion
+    ctl_minv = default_img.controllerAPIMinVersion
+    data_v = default_img.dataFormatVersion
+    data_minv = default_img.dataFormatMinVersion
+    engine_upgrade_image = common.get_upgrade_test_image(cli_v, cli_minv,
+                                                         ctl_v, ctl_minv,
+                                                         data_v, data_minv)
+
+    compatible_img = client.create_engine_image(image=engine_upgrade_image)
+    compatible_img_name = compatible_img.name
+    ei_status_value = get_engine_image_status_value(client,
+                                                    compatible_img_name)
+    compatible_img = wait_for_engine_image_state(client,
+                                                 compatible_img_name,
+                                                 ei_status_value)
+    assert compatible_img.refCount == 0
+    assert compatible_img.noRefSince != ""
+
+    return default_img, default_img_name, engine_upgrade_image, \
+        compatible_img, compatible_img_name
+
+
+def check_replica_engine(volume, engineimage):
+
+    for replica in volume.replicas:
+        if volume["state"] == "attached":
+            if volume["robustness"] == "degraded":
+                if replica.running is True:
+                    assert replica.engineImage == engineimage
+                    assert replica.currentImage == engineimage
+            else:
+                assert replica.running is True
+                assert replica.engineImage == engineimage
+                assert replica.currentImage == engineimage
```

Comment:
How about this?
```suggestion
            if volume["robustness"] != "degraded":
                assert replica.running is True

            if replica.running is True:
                assert replica.engineImage == engineimage
                assert replica.currentImage == engineimage
```

Reply:
Hi @c3y1huang , thanks for the suggestion, it's very helpful to me